OpenSim/Simulation/Control/TrackingController.cpp

```diff
@@ -76,7 +77,9 @@ void TrackingController::setNull()
 //=============================================================================
 void TrackingController::setDesiredStatesStorage(const Storage* aYDesStore)
 {
-    _desiredStatesStorage = aYDesStore;
+    auto* old = _desiredStatesStorage.release();
+    delete old;
```

Comment:
Despite Tom's excitement, I don't think this is what we should do. `_desiredStatesStorage` is a non-owning pointer and it's effectively being used as an owning pointer now. If the memory management has changed, then `_desiredStatesStorage` should be changed to a `SimTK::ResetOnCopy<std::unique_ptr<Storage>`, or something like that.

Did you discover a leak? If so, where?

I think it would be good to also edit the doxygen for this method to mention the memory management.